### PR TITLE
Feature/cid 2732/authenticate to ghe

### PR DESCRIPTION
**Pull request by mohamedlajmileanix**

### CODE REVIEW

File: src/main/kotlin/net/leanix/githubagent/GithubAgentApplication.kt [hunk 1]
```diff
@@ -1,9 +1,13 @@
 package net.leanix.githubagent
 
 import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.boot.runApplication
+import org.springframework.cloud.openfeign.EnableFeignClients
 
 @SpringBootApplication
+@EnableFeignClients(value = ["net.leanix.githubagent.client"])
```
Comment (henriqamaral) [inline src/main/kotlin/net/leanix/githubagent/GithubAgentApplication.kt:9]: I'm not sure if you need add the value for this annotation here("net.leanix.githubagent.client")

File: src/main/kotlin/net/leanix/githubagent/services/GitHubEnterpriseService.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    fun verifyJwt(jwt: String) {
+        runCatching {
+            val githubApp = githubClient.getApp("Bearer $jwt")
+            validateGithubAppResponse(githubApp)
+            logger.info("Authenticated as GitHub App: '${githubApp.name}'")
+        }.onFailure {
+            when (it) {
+                is GithubAppInsufficientPermissionsException -> throw it
+                else -> throw UnableToConnectToGithubEnterpriseException("Failed to verify JWT token")
```
Comment (geoandri) [inline src/main/kotlin/net/leanix/githubagent/services/GitHubEnterpriseService.kt:27]: Wouldn't be useful to log the exception details for easier debugging here?

File: src/main/kotlin/net/leanix/githubagent/services/CachingService.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import jakarta.annotation.PostConstruct
+import net.leanix.githubagent.config.GithubEnterpriseProperties
+import org.springframework.stereotype.Service
+
+@Service
+class CachingService(
+    private val githubEnterpriseProperties: GithubEnterpriseProperties
+) {
+    private val cache = HashMap<String, String?>()
```
Comment (henriqamaral) [inline src/main/kotlin/net/leanix/githubagent/services/CachingService.kt:11]: Have you tried one of these caches here? https://docs.spring.io/spring-boot/reference/io/caching.html

Comment (geoandri) [reply src/main/kotlin/net/leanix/githubagent/services/CachingService.kt:11]: Is this really a cache? I mean, does the cached values eventually expire? Maybe the naming is a bit confusing.

Comment (geoandri) [reply src/main/kotlin/net/leanix/githubagent/services/CachingService.kt:11]: Ok, got it now, you just need to save somewhere the token.

File: src/main/kotlin/net/leanix/githubagent/services/CachingService.kt [hunk 4]
```diff
@@ -8,19 +11,51 @@ import org.springframework.stereotype.Service
...
+                return value.expiry ?: Long.MAX_VALUE
+            }
+
+            override fun expireAfterUpdate(
+                key: String,
+                value: CacheValue,
+                currentTime: Long,
+                currentDuration: Long
+            ): Long {
+                return value.expiry ?: Long.MAX_VALUE
```
Comment (geoandri) [inline src/main/kotlin/net/leanix/githubagent/services/CachingService.kt:34]: Isn't the expiration value related to the token expiration date? How does this work? Do we need to take into account the token validity period? What happens if the token is expired?

**Review by henriqamaral — APPROVED** [review]

**Pull request merged**